TaskManager.js

```diff
@@ -119,8 +119,45 @@ class TaskManager {
    findTaskById(taskId) {
       return this.tasks.find((task) => task.id === taskId);
    }
-}
 
+   showExpiredTasks() {
+      
+      const currentDate = new Date();
+      const expiredTasks = this.tasks.filter((task) => !task.completed && task.deadline && new Date(task.deadline) < currentDate);
+      
+      console.log('Expired tasks:');
+      
+      expiredTasks.forEach((task) => {
+         
+         console.log(`  ID: ${task.id}`);
+         console.log(`  Title: ${task.title}`);
+         console.log(`  Description: ${task.description}`);
+         console.log(`  Deadline: ${task.deadline}`);
+         console.log('----------------------------');
+      
+      });
+   }
+   
+   showPendingTasks() {
 
+      const pendingTasks = this.tasks.filter((task) => !task.completed);
+      
+      pendingTasks.sort((a, b) => {
+         return new Date(a.deadline) - new Date(b.deadline);
+      });
+      
+      console.log('Pending tasks (sorted by deadline):');
```

Comment:
- Where condition, when empty list for pending tasks?
- If the task doesn't have a field "deadline", what will happen?
Think about it. 🤓

Reply:
No problem